AuthenticatorBridgeKit/SharedCryptographyService.swift

```diff
@@ -0,0 +1,143 @@
+    ///
+    /// - Parameter sharedKeychainRepository: the `SharedKeyRepository` to obtain the shared Authenticator
+    ///     key to use in encrypting/decrypting
+    ///
+    public init(sharedKeychainRepository: SharedKeychainRepository) {
+        self.sharedKeychainRepository = sharedKeychainRepository
+    }
+
+    // MARK: Methods
+
+    public func decryptAuthenticatorItems(
+        _ items: [AuthenticatorBridgeItemDataModel]
+    ) async throws -> [AuthenticatorBridgeItemDataView] {
+        let key = try await sharedKeychainRepository.getAuthenticatorKey()
+        let symmetricKey = SymmetricKey(data: key)
+
+        return items.map { item in
+            AuthenticatorBridgeItemDataView(
+                favorite: item.favorite,
+                id: item.id,
+                name: (try? decrypt(item.name, withKey: symmetricKey)) ?? "",
+                totpKey: try? decrypt(item.totpKey, withKey: symmetricKey),
+                username: try? decrypt(item.username, withKey: symmetricKey)
+            )
+        }
+    }
+
+    public func encryptAuthenticatorItems(
+        _ items: [AuthenticatorBridgeItemDataView]
+    ) async throws -> [AuthenticatorBridgeItemDataModel] {
+        let key = try await sharedKeychainRepository.getAuthenticatorKey()
+        let symmetricKey = SymmetricKey(data: key)
+
+        return items.map { item in
+            AuthenticatorBridgeItemDataModel(
+                favorite: item.favorite,
+                id: item.id,
+                name: encrypt(item.name, withKey: symmetricKey) ?? "",
+                totpKey: encrypt(item.totpKey, withKey: symmetricKey),
+                username: encrypt(item.username, withKey: symmetricKey)
+            )
+        }
+    }
+
+    /// Decrypts a string given a key.
+    ///
+    /// - Parameters:
+    ///   - string: The string to decrypt.
+    ///   - key: The key to decrypt with.
+    /// - Returns: A decrypted string, or `nil` if the passed-in string was not encoded in Base64.
+    ///
+    private func decrypt(_ string: String?, withKey key: SymmetricKey) throws -> String? {
+        guard let string, !string.isEmpty, let data = Data(base64Encoded: string) else {
+            return nil
+        }
+        let encryptedSealedBox = try AES.GCM.SealedBox(
+            combined: data
+        )
+        let decryptedBox = try AES.GCM.open(
+            encryptedSealedBox,
+            using: key
+        )
+        return String(data: decryptedBox, encoding: .utf8)
+    }
+
+    /// Encrypt a string with the given key.
+    ///
+    /// - Parameters:
+    ///   - plainText: The string to encrypt
+    ///   - key: The key to use to encrypt the string
+    /// - Returns: An encrypted string or `nil` if the string was nil
+    ///
+    private func encrypt(_ plainText: String?, withKey key: SymmetricKey) -> String? {
+        guard let plainText else {
+            return nil
+        }
+
+        let nonce = randomData(lengthInBytes: 12)
```

Comment:
🤔 "Nonce" is a problematic term due to its meaning in the UK, but unfortunately the cryptography community still uses it. It's probably moot here (the library function is called that, and we're *clearly* in cryptography land) so I mostly just wanted to sigh about it and how language is hard.